### ISSUE TITLE
NT_xent_loss the lable is Zero ?

### ISSUE THREAD

**Issue by Johnson-yue** [orpatashnik/BalaGAN#3]
https://github.com/orpatashnik/BalaGAN/blob/a34fba9f1aea3887330e9b958cf009b0683d1793/networks/encoder.py#L109

is not torch.range() ?

**Comment by orpatashnik** [comment]
Hi @Johnson-yue ,

Note that the first column of `logits` is the tensor `positives`, that is the similarity between the positive pairs. Therefore, in the `i`-th row of `logits` we have the similarity score between `z_i` and all the other representations. In each row, the similarity score which goes to the numerator (the 'positive' one) is in `logits[i][0]` and therefore we set the label to be `0` when we calculate the cross-entropy loss. 
I hope my explnation is clear, please let me know if it doesn't and I will try to elaborate.

**Comment by Johnson-yue** [comment]
yes， thank you， I got it
